### PR TITLE
Add short-circuit hook in GP_Original::closest_original.

### LINKED ISSUE #1204: Add short-circuit hook in GP_Original::closest_original
This would allow us to swap string similarity algorithm. The hook can be placed early in the `closes_original()`

https://github.com/GlotPress/GlotPress-WP/blob/6d86cc2c37515a482633220b1501bd588fefd43c/gp-includes/things/original.php#L397-L401

with something like:

```php
$pre = apply_filters( 'pre_gp_original_closest_original', false, $input, $other_strings ); 
if ( false !== $pre ) {
    return $pre;
}

```

### PR DESCRIPTION
**Pull request by gedex**
Fixes #1204.

### CODE REVIEW

**Review by rcstr — DISMISSED** [review]

File: gp-includes/things/original.php [hunk 1]
```diff
@@ -395,6 +395,21 @@ public function priority_by_name( $name ) {
 	}
 
 	public function closest_original( $input, $other_strings ) {
+		/**
+		 * Filters the preemptive return value of closest original check.
+		 *
+		 * @since x.y.z
```
Comment (ocean90) [inline gp-includes/things/original.php]:
```suggestion
		 * @since 3.0.0
```

File: gp-includes/things/original.php [hunk 2]
```diff
@@ -395,6 +395,21 @@ public function priority_by_name( $name ) {
...
+		 *
+		 * @param mixed  $pre           A preemptive return value of closest original
+		 *                              check. Default false.
```
Comment (ocean90) [inline gp-includes/things/original.php]:
```suggestion
		 * @param string|false|null  $pre A preemptive return value of closest original
		 *                                check. Default false.
```

File: gp-includes/things/original.php [hunk 3]
```diff
@@ -395,6 +395,21 @@ public function priority_by_name( $name ) {
...
+		 * @param string $input         Input string.
+		 * @param array  $other_strings List of string to check against input string.
```
Comment (ocean90) [inline gp-includes/things/original.php]:
```suggestion
		 * @param array  $other_strings List of strings to check against input string.
```

**Review by ocean90 — CHANGES_REQUESTED** [review]
Thanks for the pull request! Let's clarify the supported types and update the @since tag.

**Comment by gedex** [comment]
Thanks for the feedback, @ocean90.

**Pull request merged**